### PR TITLE
Bit numbering

### PR DESCRIPTION
**Pull request by mortenvp**

### CODE REVIEW

File: src/bitter/endian_field_offset.hpp [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+///
+/// We would have the following layout of the four fields inside the byte:
+///
+///     0   1   2   3   4   5   6   7
+///   +---+-------+-----------+-------+
+///   | 0 | 1   0 | 1   1   1 | 0   0 |
+///   +---+-------+-----------+-------+
+///     ^
+///     |             most significant
+///     +-----------+ bit
```
Comment (jpihl) [inline src/bitter/endian_field_offset.hpp]: I would prefer if this only documented once either here or in the Readme, and then referenced

File: README.rst [hunk 2]
```diff
@@ -44,6 +49,102 @@ above lets read back the four bit fields from the uint32_t value::
...
+    +---------------------------------------------------------------+
+    |Version|  IHL  |Type of Service|          Total Length         |
+    +---------------------------------------------------------------+
+
+Lets look at how we work with this in bitter. In bitter the fields
+laid out from bit number 0.
+
+So imagine we have the following bit reader::
+
+    auto reader = bitter::lsb_reader<uint8_t, 1, 2, 3, 2>(0xdeadbeef);
```
Comment (jpihl) [inline README.rst]: It might be a little confusing that the value is 64 bit. Maybe shorten to 0xde

File: src/bitter/lsb0_reader.hpp [hunk 3]
```diff
@@ -0,0 +1,15 @@
+// Copyright (c) Steinwurf ApS 2016.
+// All Rights Reserved
+//
+// Distributed under the "BSD License". See the accompanying LICENSE.rst file.
+#pragma once
+
+#include "reader.hpp"
+
+namespace bitter
```
Comment (jpihl) [inline src/bitter/lsb0_reader.hpp]: Why have this typedef?

File: test/src/test_reader.cpp [hunk 4]
```diff
@@ -3,7 +3,8 @@
 //
 // Distributed under the "BSD License". See the accompanying LICENSE.rst file
 
-#include <bitter/reader.hpp>
+#include <bitter/lsb0_reader.hpp>
+#include <bitter/msb0_reader.hpp>
```
Comment (jpihl) [inline test/src/test_reader.cpp]: I would prefer to have two separate test files for the two readers (if they are to be split like this, that is).

File: test/src/test_readme.cpp [hunk 5]
```diff
@@ -3,39 +3,73 @@
...
+{
+    auto reader = bitter::lsb0_reader<uint32_t, 8, 8, 8, 8>(0x12345678);
+
+    uint8_t value0 = reader.field<0>().read_as<uint8_t>(); // Read bits 0-7
+    uint8_t value1 = reader.field<1>().read_as<uint8_t>(); // Read bits 8-15
+    uint8_t value2 = reader.field<2>().read_as<uint8_t>(); // Read bits 16-23
+    uint8_t value3 = reader.field<3>().read_as<uint8_t>(); // Read bits 24-31
 
-    assert(writer.data() == 0xdeadbeef);
+    assert(value0 == 0x78);
```
Comment (jpihl) [inline test/src/test_readme.cpp]: Why assert and not EXPECT_EQ or ASSERT_EQ

File: src/bitter/lsb0_reader.hpp (hunk 3 above)
Comment (mortenvp) [reply src/bitter/lsb0_reader.hpp]: It is just a convenience alias, in many cases you would simply use one or the other. So no need for the template argument. In cases where you need a reader that can work in both modes you can use the reader class directly.

**Comment by jpihl** [comment]
looks good!

**Comment by mortenvp** [comment]
@jpihl should we merge?

**Comment by jpihl** [comment]
Sounds like a good idea

**Pull request merged**